### PR TITLE
Geo dependencies optional

### PR DESCRIPTION
**Pull request by doctrino**
## Description
I have moved `shapely `and `geopandas `as optional dependencies, available with the option `geo`.

## Checklist:
- [x] Tests added/updated. Not relevant. 
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change. 
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring. 
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) per [semantic versioning](https://semver.org/).

### CODE REVIEW

**Review by erlendvollset — CHANGES_REQUESTED** [review]

File: setup.py [hunk 1]
```diff
@@ -21,9 +21,13 @@
         "requests>=2.21.0,<3.0.0",
         "pandas",
         "requests_oauthlib==1.*",
-        "geopandas==0.10.*",
-        "shapely==1.*",
     ],
+    extras_require={
```
[the remark this answers is not in the document]
Comment (doctrino) [reply setup.py]: Done. Moved to setup-core.

File: README.md [hunk 2]
```diff
@@ -38,6 +38,11 @@ To install this package:
 $ pip install cognite-sdk
 ```
 
+To install with geopandas and shapely support.
+```bash
+$ pip install cognite-sdk[geo]
+```
```
[the remark this answers is not in the document]
Comment (doctrino) [reply README.md]: Done.

**Review by erlendvollset — APPROVED** [review]
LGTM 🚀

File: tests/tests_integration/test_api/test_transformations/test_notifications.py [hunk 3]
```diff
@@ -59,6 +59,7 @@ def new_notification_by_external_id(cognite_client, new_transformation):
 
 
 class TestTransformationNotificationsAPI:
+    @pytest.mark.xfail
```
Comment (erlendvollset) [inline tests/tests_integration/test_api/test_transformations/test_notifications.py:62]: You can add this annotation on the class itself.
i.e.
```
@pytest.mark.xfail
class TestTransformationNotificationsApi:
    ...
```

Comment (erlendvollset) [reply tests/tests_integration/test_api/test_transformations/test_notifications.py:62]: or on a module by doing
pytestmarks = [pytest.mark.xfail]

**Pull request merged**